be/src/http/action/stream_load.cpp

```diff
@@ -237,6 +242,9 @@ Status StreamLoadAction::_on_header(HttpRequest* http_req, StreamLoadContext* ct
             ss << "body exceed max size: " << max_body_bytes << ", limit: " << max_body_bytes;
             return Status::InternalError(ss.str());
         }
+
+        // allocate buffer in advance.
+        ctx->buf = ByteBuffer::allocate(ctx->body_bytes);
```

Comment:
If this is not a json file, there is no need to allocate a buffer of size `body_bytes`, it would be a waste of memory.

Reply:
Allocate extras SIMDJSON_PADDING bytes if this is a JSON file for better performance.

Reply:
The CSV format load request may also provide `Content-Length`, so this is necessary.

Reply:
But CSV does not need to buffer all data.

Reply:
Got it. This allocation now only work for JSON.